Helper/Cart.php

```diff
@@ -2952,4 +2954,27 @@ public function isCollectDiscountsByPlugin($quote)
             || ($this->configHelper->isActive($quote->getStore()->getId())
             && version_compare($this->configHelper->getStoreVersion(), '2.3.4', '<'));
     }
+    
+    /**
+     * Get SKU of quote item
+     *
+     * @param \Magento\Quote\Model\Quote\Item $item
+     *
+     * @return string
+     */
+    public function getSkuFromQuoteItem($item)
+    {
+        // Get "Dynamic SKU" of the bundle product, so the quote item can be recognized.
+        if (($product = $item->getProduct())
+            && $item->getProductType() == \Magento\Catalog\Model\Product\Type::TYPE_BUNDLE
```

Comment:
What if we refactor this a little and do an early return if the type is not "bundle"? In this case, we do not need to get product for majority of cases.

Reply:
Make sense. Done